lineapy/cli/cli.py

```diff
@@ -428,10 +428,17 @@ def python_cli(
             )
             exit(1)
 
-        ap = AirflowPlugin(db, tracer.tracer_context.get_session_id())
+        artifact_collection = ArtifactCollection(db, slice)
         task_dependencies = ast.literal_eval(airflow_task_dependencies or "{}")
-        _, task_graph = extract_taskgraph(slice, task_dependencies)
-        ap.sliced_airflow_dag(export_slice_to_airflow_dag, task_graph)
+
+        # Construct pipeline writer
+        pipeline_writer = AirflowPipelineWriter(
+            artifact_collection=artifact_collection,
+            dependencies=task_dependencies,
+        )
+
+        # Write out pipeline files
+        pipeline_writer.write_pipeline_files()
```

Comment:
we should use the pipeline object as entry point here. not super urgent i think because cli entrypoints for non-airflow pipelines isnt very well defined. can either tackle this in this PR or leave as a TODO for later leaving up to you.

Reply:
LIN-319 design doc to capture the expected behavior here. suggest not doing it in this PR.

Reply:
Adding a TODO note in code.